File: system-test/typescript/src/testkit/driver/port/external/clock/clock-driver.ts
import { Result } from '../../../../common/result.js';
import {
  ClockErrorResponse,
  GetTimeResponse,
  ReturnsTimeRequest,
} from '../../../../common/dtos.js';

export interface ClockDriver {
  goToClock(): Promise<Result<void, ClockErrorResponse>>;
  getTime(): Promise<Result<GetTimeResponse, ClockErrorResponse>>;
  returnsTime(request: ReturnsTimeRequest): Promise<Result<void, ClockErrorResponse>>;
  close(): Promise<void>;
}

File: system-test/java/src/main/java/com/mycompany/myshop/testkit/driver/port/external/clock/ClockDriver.java
package com.mycompany.myshop.testkit.driver.port.external.clock;

import com.mycompany.myshop.testkit.driver.port.external.clock.dtos.GetTimeResponse;
import com.mycompany.myshop.testkit.driver.port.external.clock.dtos.ReturnsTimeRequest;
import com.mycompany.myshop.testkit.driver.port.external.clock.dtos.error.ClockErrorResponse;
import com.mycompany.myshop.testkit.common.Result;

public interface ClockDriver extends AutoCloseable {
    Result<Void, ClockErrorResponse> goToClock();
    Result<GetTimeResponse, ClockErrorResponse> getTime();
    Result<Void, ClockErrorResponse> returnsTime(ReturnsTimeRequest request);
}
